From 8d2dfbcddc81722d5b3d5568395618256f45063d Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 9 Mar 2017 11:04:04 +0800
Subject: [PATCH 1643/5242] MLK-14299-1 video: mxsfb: clear the overlay fb
 memory buffer

commit  ee60f6d0cc5292e117e139fe47bdbf25d6569dee from
https://source.codeaurora.org/external/imx/linux-imx.git

Clear the overlay fb memory buffer after allocated to avoid
random pixel data in it before using it.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 6b1a31b8fc39c7cf19ff3dcdc57f2f4059632f45)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 57722ad..2db0638 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -2032,6 +2032,9 @@ static int mxsfb_overlay_map_video_memory(struct mxsfb_info *fbi,
 		return -ENOMEM;
 	}
 
+	/* clear overlay fb memory buffer */
+	memset(ofb->video_mem, 0x0, ofb->video_mem_size);
+
 	ofb->ol_fb->fix.smem_start = ofb->video_mem_phys;
 	ofb->ol_fb->fix.smem_len   = ofb->video_mem_size;
 	ofb->ol_fb->screen_base    = ofb->video_mem;
-- 
1.7.9.5

